codec-http/src/main/java/io/netty/handler/codec/http/HttpObjectDecoder.java

```diff
@@ -587,7 +591,7 @@ private State readHeaders(ByteBuf buffer) {
 
         State nextState;
 
-        if (isContentAlwaysEmpty(message)) {
+        if (isContentAlwaysEmpty(message) || HttpUtil.hasEmptyContent(message)) {
```

Comment:
@jza70 can you just add another unit test that shows the original `Content-Length: 0` header is preserved ?